amp-pwa-reader/src/js/Article.js

```diff
@@ -0,0 +1,250 @@
+class Article {
+
+  constructor(url, card) {
+    this.url = shadowReader.backend.getAMPUrl(url);
+    this.card = card;
+    Article.articles[this.url] = this;
+  }
+
+  fetch() {
+
+    // unfortunately fetch() does not support retrieving documents,
+    // so we have to resort to good old XMLHttpRequest.
+    var xhr = new XMLHttpRequest();
+
+    return new Promise((resolve, reject) => {
+      xhr.open('GET', 'https://seed-octagon.glitch.me/' + encodeURIComponent(this.url), true);
+      xhr.responseType = 'document';
+      xhr.setRequestHeader('Accept', 'text/html');
+      xhr.onload = () => {
+        var isAMP = xhr.responseXML.documentElement.hasAttribute('amp') || xhr.responseXML.documentElement.hasAttribute('⚡');
+        if (isAMP) {
+          resolve(xhr.responseXML);
+        } else {
+          reject('Article does not have an AMP version.');
+        }
+      }; // .responseXML contains a ready-to-use Document object
+      xhr.send();
+    });
+
+  }
+
+  load() {
+    return (this.doc ? new Promise(resolve => resolve()) : this.fetch().then(doc => {
```

Comment:
Could be written as:
`return (this.doc ? Promise.resolve() : this.fetch().then(doc => {`